bindgen/private/bindgen.bzl

```diff
@@ -124,7 +124,9 @@ def _generate_cc_link_build_info(ctx, cc_lib):
                 linker_search_paths.append(lib.pic_static_library.dirname)
                 compile_data.append(lib.pic_static_library)
 
-        linker_flags.extend(linker_input.user_link_flags)
+        if linker_input.user_link_flags:
```

Comment:
@UebelAndre I think this will only work if we link using `rustc`. If we run the linker directly (i.e. enabling [experimental_use_cc_common_link](https://github.com/bazelbuild/rules_rust/blob/012af3ba8fda6acb91a440f8c86b5a45d2c4e64e/rust/settings/BUILD.bazel#L69), it won't work because the linker doesn't understand `-C link-args`. Only rustc understands `-C link-args=<args>` and passes `<args>` to the linker.

Reply:
Another subtle thing I notice in the example provided in https://github.com/bazelbuild/rules_rust/issues/2406 is that `rust_bindgen_library` shouldn't trigger any linking (neither calling `rustc` with `-C linker=<> -C link-args` or a separate linking action) because the macro (which creates rust_library -> rust_bindgen) only produces an rlib, not a shared library or binary.

Is 
```
# BUILD
load("@rules_rust//bindgen:defs.bzl", "rust_bindgen_library")                    
                                                                                 
cc_library(                                                                      
    name = "foo",                                                                
    srcs = ["foo.c"],                                                            
    hdrs = ["foo.h"],                                                            
    linkopts = ["-framework", "Security"],                                       
)                                                                                
                                                                                 
rust_bindgen_library(                                                            
    name = "foo_bindgen",                                                        
    cc_lib = ":foo",                                                             
    header = "foo.h",                                                            
)
```
all you need to reproduce the error with the link flags? Or is there another downstream target like `rust_binary` that causes a linking?

Reply:
IIUC, the right fix isn't simply
1. If use `cc_common.link`, don't use `-C link-args=<link flags>`
2. Else, use `-C link-args`
at the bindgen rule level.

IMO the right fix is to keep link flags without `-C link-args=` as before. At the downstream (shared library or binary) targets which run linking, we then decide whether to add `-C link-args` or not based on the setting of [experimental_use_cc_common_link](https://github.com/bazelbuild/rules_rust/blob/012af3ba8fda6acb91a440f8c86b5a45d2c4e64e/rust/settings/BUILD.bazel#L69).

Reply:
More context: In https://github.com/bazelbuild/rules_rust/blob/90e4505c19e8276311cd5ad7d9d894aa659105dd/rust/private/rustc.bzl#L998-L1013. 

The rules' implementation already ignore linking when building `rlib` and `lib`. `rust_bindgen_library` produces `rlib` which shouldn't involve any link flags at all.

Reply:
I filed https://github.com/bazelbuild/rules_rust/issues/2413 to triage the issue.